packages/peregrine/lib/talons/CategoryList/useCategoryList.js

```diff
@@ -12,16 +12,19 @@ import { useLazyQuery } from '@apollo/react-hooks';
 export const useCategoryList = props => {
     const { query, id } = props;
 
-    const [runQuery, queryResult] = useLazyQuery(query);
-    const { loading, error, data } = queryResult;
+    console.log('USE CATEGORY LIST');
 
+    // Run the query immmediately.
+    const { loading, error, data, refetch } = useQuery(query, {
+        variables: {
+            id
+        }
+    });
+
+    // And every time id changes.
     useEffect(() => {
-        runQuery({
-            variables: {
-                id
-            }
-        });
-    }, [runQuery, id]);
+        refetch();
+    }, [id, refetch]);
```

Comment:
Does `refetch` see the new `id`?

Reply:
Good catch -- according to docs you need to provide new variables.

Reply:
Also after inspection this runs the initial query and then immediately runs the `refetch`. I think we want `useLazyQuery` here.